dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1036 @@
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent : ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
 
-        # image_shapes should be a non static field to prevent problems
+        # Default image_shapes should be a non static field to prevent problems
         # when multiple instances of ImageMapper are created
-        self.image_shapes = {
-            "LSTCam": (110, 110, 1),
-            "LSTSiPMCam": (234, 234, 1),
-            "FlashCam": (112, 112, 1),
-            "NectarCam": (110, 110, 1),
-            "SCTCam": (120, 120, 1),
-            "DigiCam": (96, 96, 1),
-            "CHEC": (48, 48, 1),
-            "ASTRICam": (56, 56, 1),
-            "VERITAS": (54, 54, 1),
-            "MAGICCam": (78, 78, 1),
-            "FACT": (90, 90, 1),
-            "HESS-I": (72, 72, 1),
-            "HESS-II": (104, 104, 1),
+        self.default_image_shapes = {
+            "LSTCam": 110,
+            "LSTSiPMCam": 234,
+            "FlashCam": 112,
+            "NectarCam": 110,
+            "SCTCam": 120,
+            "DigiCam": 96,
+            "CHEC": 48,
+            "ASTRICam": 56,
+            "VERITAS": 54,
+            "MAGICCam": 78,
+            "FACT": 92,
+            "HESS-I": 72,
+            "HESS-II": 104,
         }
 
         # Camera types
-        if camera_types:
-            self.camera_types = []
-            for camera_type in camera_types:
-                if camera_type in self.image_shapes:
-                    self.camera_types.append(camera_type)
-                else:
-                    logger.error("Camera type {} isn't supported.".format(camera_type))
-        else:
-            self.camera_types = [cam for cam in self.image_shapes]
-
-        # Mapping method
-        if mapping_method is None:
-            mapping_method = {}
-        self.mapping_method = {
-            **{c: "oversampling" for c in self.camera_types},
-            **mapping_method,
-        }
+        self.geometry = geometry
+        self.camera_type = self.geometry.name
+        self.image_shape = self.default_image_shapes[self.camera_type]
+        self.n_pixels = self.geometry.n_pixels
+        # Rotate the pixel positions by the pixel to align
+        self.geometry.rotate(self.geometry.pix_rotation)
+
+        self.pix_x = np.around(self.geometry.pix_x.value, decimals=3)
+        self.pix_y = np.around(self.geometry.pix_y.value, decimals=3)
+
+        self.x_ticks = np.unique(self.pix_x).tolist()
+        self.y_ticks = np.unique(self.pix_y).tolist()
+
+        # Additional smooth the ticks for 'DigiCam' and 'CHEC' cameras
+        if self.camera_type == "DigiCam":
```

Comment:
General suggestion: Is it not more convenient the use con Enum to avoid problems for camera_type or other potential types. 
Example: 

from enum import Enum
class Weekday(Enum):
    MONDAY = 1
    TUESDAY = 2
    WEDNESDAY = 3
    THURSDAY = 4
    FRIDAY = 5
    SATURDAY = 6
    SUNDAY = 7

print(Weekday.THURSDAY)